### PR TITLE
Reimplement promise.all

### PR DESCRIPTION
**Pull request by janedegtiareva**

### CODE REVIEW

File: assembly/contract.ts [hunk 1]
```diff
@@ -258,14 +258,15 @@ export class ContractPromise {
...
-  //   return { id };
-  // }
+  static all(promises: ContractPromise[]): ContractPromise {
+    assert(promises.length > 0);
+    const ids = new Array<u64>(promises.length);
+    for (let i = 1; i < promises.length; i++) {
+      ids[i] = promises[i].id;
+    }
+    const id = runtime_api.promise_and(ids as i64, ids.length);
+    return { id };
```
Comment (ailisp) [inline assembly/contract.ts]: promises[0] seems to be omitted. Is it intended.

File: assembly/contract.ts [hunk 2]
```diff
@@ -258,14 +258,15 @@ export class ContractPromise {
...
-  //   }
-  //   return { id };
-  // }
+  static all(promises: ContractPromise[]): ContractPromise {
+    assert(promises.length > 0);
+    const ids = new Array<u64>(promises.length);
+    for (let i = 1; i < promises.length; i++) {
+      ids[i] = promises[i].id;
+    }
+    const id = runtime_api.promise_and(ids as i64, ids.length);
```
Comment (ailisp) [inline assembly/contract.ts]: Does promises.length=1 case (`promise_and([], 0)`) also work? Maybe add a test case for that too?

File: assembly/contract.ts (hunk 1 above)
Comment (janedegtiareva) [reply assembly/contract.ts]: no, good catch!

**Comment by janedegtiareva** [comment]
Runs with this contract: https://github.com/nearprotocol/nearcore/pull/1295

**Pull request merged**